Source/Core/Core/HW/WiimoteEmu/Dynamics.cpp

```diff
@@ -11,47 +11,91 @@
 #include "Core/HW/Wiimote.h"
 #include "Core/HW/WiimoteEmu/WiimoteEmu.h"
 #include "InputCommon/ControllerEmu/ControlGroup/Buttons.h"
+#include "InputCommon/ControllerEmu/ControlGroup/Cursor.h"
 #include "InputCommon/ControllerEmu/ControlGroup/Force.h"
 #include "InputCommon/ControllerEmu/ControlGroup/Tilt.h"
 
-namespace WiimoteEmu
+namespace
 {
 constexpr int SHAKE_FREQ = 6;
 // Frame count of one up/down shake
 // < 9 no shake detection in "Wario Land: Shake It"
 constexpr int SHAKE_STEP_MAX = ::Wiimote::UPDATE_FREQ / SHAKE_FREQ;
 
-void EmulateShake(NormalizedAccelData* const accel, ControllerEmu::Buttons* const buttons_group,
-                  const double intensity, u8* const shake_step)
+// Given a velocity, acceleration, and maximum jerk value,
+// calculate change in position after a stop in the shortest possible time.
+// Used to smoothly adjust acceleration and come to complete stops at precise positions.
+double CalculateStopDistance(double velocity, double acceleration, double max_jerk)
```

Comment:
Are those any well-known or common algorithms to calculate this? Do you have a few keywords to search for by chance (which would make sense as accompanying comment for future readers?)

Reply:
I guess it would be called "equations for motion with constant jerk". The function is based on usages of that. I can add a comment.

Reply:
Good enough for me, I just thought thats a set of formulas from `$somewhere`.